src/Parser.py

```diff
@@ -109,13 +103,25 @@ def transpile(self):
 	def class_body(self):
 		if self.expect('pass'): return
 		static = self.expect('static')
-		if self.expect('class'): self.nested_class()
+		if self.expect('@', 'tool'): self.is_tool = True
+		elif self.expect('extends'): self.base_class = self.consume()
+		elif self.expect('class_name'): self.class_name = self.consume()
```

Comment:
these are handled differently between the script-level class and nested classes,
and your code currently overwrites script-level data with the last nested class info